addons/test_mail/tests/test_mail_template.py

```diff
@@ -128,6 +134,27 @@ def test_template_add_context_action(self):
         self.assertEqual(action.name, 'Send Mail (%s)' % self.email_template.name)
         self.assertEqual(action.binding_model_id.model, 'mail.test.simple')
 
+    @users('demo')
+    def test_mail_composer_add_signature(self):
+        create_values = {
+            'name': 'TestTemplate',
+            'subject': 'About Test',
+            'body_html': '<p>Hello Test</p>',
+            'model_id': self.env['ir.model']._get('mail.test.track').id,
+        }
+
+        template = self.env['mail.template'].create(create_values)
+        mail_id = template.send_mail(self.test_track.id, notif_layout='mail.mail_notification_paynow')
+        mail = self.env['mail.mail'].sudo().browse(mail_id)
+        user = self.env.user
+        self.test_track.user_id = user
+        base_template = self.env.ref('mail.mail_notification_paynow', raise_if_not_found=True).with_context(signature="Test")
+        mail_body = base_template._render({"signature": "Test", 'record': self.test_track, 'message': mail.mail_message_id, 'company': user.company_id}, engine='ir.qweb', minimal_qcontext=True)
```

Comment:
This is quite low level... you test that the template works, but not really the functional flow. I think it is more interesting to test the result of a post using the ``mail_notification_paynow`` template (it may be given to ``message_post`` using the ``email_layout_xmlid`` parameter that is propagated through the call chain).